### ISSUE TITLE
`Style/HashEachMethods` mistakenly reported for zipped arrays

### ISSUE THREAD

**Issue by toncid** [rubocop/rubocop#12880]
An autocorrectable `Style/HashEachMethods` offense can lead to broken code. RuboCop treats zipped arrays as a hash and therefore offers an optimization that makes no sense in this case.

--------

## Expected behavior

RuboCop accepts the call to `each` with an unused block argument on an array produced by `Array#zip`.

## Actual behavior

`Array#zip` returns an array of arrays. Switching `each` to `each_value` makes no sense.

Command output:
```
For /.../project: configuration from /.../project/.rubocop.yml
Inheriting configuration from /.../project/.rubocop_todo.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /.../project/app/my_class.rb
Loading cache from /.../.cache/rubocop_cache/fadea096a1fb3fd3f34cefa5204bbfaf11610c5d/7e0edf036d9954ae60d86a00cb9bb69ab476e3c6/bec8eaa5640eb2379d2f06895c1743dabb5ad132
C

Offenses:

app/my_class.rb:8:5: C: [Correctable] Style/HashEachMethods: Use each_value instead of each and remove the unused _v1 block argument.
    zipped_array.each do |_v1, v2| ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

## Steps to reproduce the problem

```ruby
class MyClass
  def do_something
    array1 = [:a1, :b1, :c1]
    array2 = [:a2, :b2, :c2]

    zipped_array = array1.zip(array2)

    zipped_array.each do |_v1, v2|  # Style/HashEachMethods offense
      puts ">>> #{v2}"
    end
  end
end
```

## RuboCop version

```
1.63.4 (using Parser 3.3.1.0, rubocop-ast 1.31.3, running on ruby 3.2.2) +server [arm64-darwin23]
  - rubocop-factory_bot 2.25.1
  - rubocop-performance 1.21.0
  - rubocop-rails 2.24.1
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.29.1
  - rubocop-rspec_rails 2.28.3
```

**Comment by vlad-pisanov** [comment]
This cop is already [marked as unsafe](https://docs.rubocop.org/rubocop/cops_style.html#safety-24), since it can't tell whether the receiver is an array or a hash.

**Comment by toncid** [comment]
Ah, missed this detail. 🤦‍♂️  Thanks!
